compiler/rustc_typeck/src/check/generator_interior.rs

```diff
@@ -210,6 +224,40 @@ impl<'a, 'tcx> Visitor<'tcx> for InteriorVisitor<'a, 'tcx> {
         NestedVisitorMap::None
     }
 
+    fn visit_arm(&mut self, arm: &'tcx Arm<'tcx>) {
+        if arm.guard.is_some() {
+            self.guard_bindings.push(<_>::default());
+        }
+        self.visit_pat(&arm.pat);
+        if let Some(ref g) = arm.guard {
```

Comment:
instead of accessing the arm's fields directly, could you destructure it so if new fields are added, this gets updated? e.g. `let Arm { guard, pat } = arm;`

This code is duplicating what normally goes in `super_visit_arm()` so it's helpful to have static checks like this.

Reply:
That is a good idea. I also updated other field accesses to `arm` towards the end of this function. There are other fields in `arm` such as `hir_id`, but I am going to elide them for now.

Reply:
If you use `..` it sort of defeats the purpose of what I was suggesting. I'd rather list out all fields and assign them to `_`.